src/cacheHttpClient.ts

```diff
@@ -144,7 +144,33 @@ export async function downloadCache(
     const stream = fs.createWriteStream(archivePath);
     const httpClient = new HttpClient("actions/cache");
     const downloadResponse = await httpClient.get(archiveLocation);
+
+    // Abort download if no traffic received over the socket.
+    downloadResponse.message.socket.setTimeout(SocketTimeout, () => {
+        downloadResponse.message.destroy();
```

Comment:
Nodejs newbie here but do we need to manually destroy the socket?

Reply:
Yes, from the [docs](https://nodejs.org/api/net.html#net_socket_settimeout_timeout_callback), this only invokes the timeout callback.  `end()` or `destroy()` must be called manually.  During testing, I found `end()` wasn't sufficient to kill the connection.  From the docs it says `Half-closes the socket. i.e., it sends a FIN packet. It is possible the server will still send some data.`, so if the server is unresponsive it might not close the socket.